src/sentry/sentry_metrics/querying/data/execution.py

```diff
@@ -335,6 +386,26 @@ def __init__(self, organization: Organization, referrer: str):
         # Tracks the number of queries that have been executed (for measuring purposes).
         self._number_of_executed_queries = 0
 
+        # We load the blocked metrics for the supplied projects.
+        self._blocked_metrics_for_projects = self._load_blocked_metrics_for_projects()
+
+    def _load_blocked_metrics_for_projects(self) -> Mapping[str, Set[int]]:
+        """
+        Load the blocked metrics for the supplied projects and stores them in the executor in an efficient way that
+        speeds up the determining of the projects to exclude from the query.
+        """
+        blocked_metrics_for_projects: Dict[str, Set[int]] = {}
```

Comment:
This internal repr allows us to very efficiently determine the final set of projects to query.